HISTORY.md

```diff
@@ -1,3 +1,16 @@
+# Release 0.24.0
+
+* TileDB-Py 0.24.0 includes TileDB Embedded [2.18.0](https://github.com/TileDB-Inc/TileDB/releases/tag/2.18.0)
```

Comment:
TBH I'd be happy if we deferred 0.24.0 until 2.18.2

Otherwise we'll need to do this all over again

Just my $0.02, not sure if that would be acceptable to everyone